nerfstudio/exporter/marching_cubes.py

```diff
@@ -0,0 +1,249 @@
+    """
+    Evaluate a signed distance function (SDF) for a batch of points.
+
+    Args:
+        sdf: A callable function that takes a tensor of size (N, 3) containing
+            3D points and returns a tensor of size (N,) with the SDF values.
+        points: A torch tensor containing 3D points.
+
+    Returns:
+        A torch tensor with the SDF values evaluated at the given points.
+    """
+    z = []
+    for _, pnts in enumerate(torch.split(points, 100000, dim=0)):
+        z.append(sdf(pnts))
+    z = torch.cat(z, axis=0)
+    return z
+
+
+def evaluate_multiresolution_sdf(
+    evaluate: Callable,
+    points_pyramid: List[TensorType[3, "height", "width", "depth"]],
+    coarse_mask: Union[TensorType[1, 1, "height", "width", "depth"], None],
+    x_max: float,
+    x_min: float,
+    crop_n: int,
+) -> TensorType["batch"]:
+
+    """
+    Evaluate SDF values using a multi-resolution approach with a given point pyramid.
+
+    Args:
+        evaluate: A callable function to evaluate SDF values at given points.
+        points_pyramid: A list of torch tensors representing points at different resolutions.
+        coarse_mask: A binary mask tensor indicating regions of the bounding box where the SDF
+            is expected to have a zero-crossing.
+        x_max: The maximum x-coordinate of the bounding box.
+        x_min: The minimum x-coordinate of the bounding box.
+        crop_n: The resolution of the grid used to sample the SDF.
+
+    Returns:
+        A torch tensor with the SDF values evaluated at the given points.
+    """
+    mask = None
+    threshold = 2 * (x_max - x_min) / crop_n * 8
+    for pid, pts in enumerate(points_pyramid):
+        coarse_n = pts.shape[-1]
+        pts = pts.reshape(3, -1).permute(1, 0).contiguous()
+
+        if mask is None:
+            # Only evaluate SDF
+            if coarse_mask is not None:
+                pts_sdf = torch.ones_like(pts[:, 1])
+                valid_mask = torch.nn.functional.grid_sample(coarse_mask, pts[None, None, None])[0, 0, 0, 0] > 0
+                if valid_mask.any():
+                    pts_sdf[valid_mask] = evaluate(pts[valid_mask].contiguous())
+            else:
+                pts_sdf = evaluate(pts)
+        else:
+            mask = mask.reshape(-1)
+            pts_to_eval = pts[mask]
+
+            if pts_to_eval.shape[0] > 0:
+                pts_sdf_eval = evaluate(pts_to_eval.contiguous())
+                pts_sdf[mask] = pts_sdf_eval
+
+        if pid < 3:
+            # Update mask
+            mask = torch.abs(pts_sdf) < threshold
+            mask = mask.reshape(coarse_n, coarse_n, coarse_n)[None, None]
+            mask = upsample(mask.float()).bool()
+
+            pts_sdf = pts_sdf.reshape(coarse_n, coarse_n, coarse_n)[None, None]
+            pts_sdf = upsample(pts_sdf)
+            pts_sdf = pts_sdf.reshape(-1)
+
+        threshold /= 2.0
+    return pts_sdf
+
+
+@torch.no_grad()
+def get_surface_sliding(
+    sdf: Callable,
```

Comment:
Regarding my comments on "isosurface threshold", `sdf` can be renamed to some name which is a callable function. It'll be sdf for an sdf-based method but densities for a nerf-based method.